pkg/analyser/analyzer.go

```diff
@@ -42,18 +42,17 @@ type AnalyzerOptions struct {
 }
 
 type Analyzer struct {
-	alerter *alerter.Alerter
-	options AnalyzerOptions
-	filter  filter.Filter
+	analysis *Analysis
+	alerter  *alerter.Alerter
+	options  AnalyzerOptions
+	filter   filter.Filter
 }
 
 func NewAnalyzer(alerter *alerter.Alerter, options AnalyzerOptions, filter filter.Filter) *Analyzer {
-	return &Analyzer{alerter, options, filter}
+	return &Analyzer{NewAnalysis(options), alerter, options, filter}
 }
 
-func (a Analyzer) Analyze(remoteResources, resourcesFromState []*resource.Resource) (Analysis, error) {
-	analysis := Analysis{options: a.options}
-
+func (a Analyzer) CompareEnumeration(remoteResources, resourcesFromState []*resource.Resource) *Analysis {
```

Comment:
Ok I find this new interface quite confusing and it took me some time understanding how that will work and why you needed this.

As it occurs you need 2 separate step to first get resource that need to be read and then read them if in deep mode, I thinks we could make the analyzer interface more comprehensive:
- I don't really like that the 2 calls return the pointer to an internal analyze I think you should create a new one at each step and init the second one with either the internal one or an argument to the second step ?
- It's strange that you need to call a method to get the analysis correct (analyzer.SetAlerts()) and that this work only because you returned a pointer to the internal analysis that will be updated after being returned.